### ISSUE TITLE
Is it possible to load all survey choices by url once for the survey?

### ISSUE THREAD

**Issue by nathan-cv** [surveyjs/survey-library#4920]
Rather than have multiple calls to an api to retrieve the options for each question, I'd like to set the options at the survey level, then specify the path for each question.

I thought that it would cache the choicesByUrl since the same for each question, but that doesn't seem to be the case.

Thanks!

**Comment by andrewtelnov** [comment]
@nathan-cv by default, SurveyJS cache choices with the same URL/parameters. It means that on the first load it will go to the cache and then it will be used in all other questions without requesting the server.

Thank you,
Andrew

**Comment by nathan-cv** [comment]
Thanks, @andrewtelnov.  That's the behavior I expected.  Would having the same url, but different path properties set for each choicesByUrl be considered the same URL/parameters?

My endpoint returns all the choices in one response.  The survey is requesting the data from the same URL multiple times.  Maybe all subsequent requests start before any of the results are in the cache?

Thanks,
Nathan

**Comment by andrewtelnov** [comment]
@nathan-cv Do you have the same final url? We do not send the same request if it has been already sent. The next request is added into a list and get the data after the request is finished.

Thank you,
Andrew

**Comment by nathan-cv** [comment]
@andrewtelnov yes, same final request.  I'd expect that an awaitable would be returned if there's already a request for the same URL/parameters in progress.
